MapboxCoreNavigationTests/PassiveLocationDataSourceTests.swift

```diff
@@ -0,0 +1,103 @@
+    static func -(a: CLLocationCoordinate2D, b: CLLocationCoordinate2D) -> CLLocationCoordinate2D {
+        return CLLocationCoordinate2D(latitude: a.latitude - b.latitude, longitude: a.longitude - b.longitude)
+    }
+
+    func length() -> Double {
+        return sqrt(latitude * latitude + longitude * longitude)
+    }
+}
+
+class PassiveLocationDataSourceTests: XCTestCase {
+    class Delegate: PassiveLocationDataSourceDelegate {
+        let road: Road
+        let locationUpdateExpectation: XCTestExpectation
+        
+        init(road: Road, locationUpdateExpectation: XCTestExpectation) {
+            self.road = road
+            self.locationUpdateExpectation = locationUpdateExpectation
+        }
+        
+        func passiveLocationDataSource(_ dataSource: PassiveLocationDataSource, didUpdateLocation location: CLLocation, rawLocation: CLLocation) {
+            print("Got location: \(rawLocation.coordinate.latitude), \(rawLocation.coordinate.longitude) → \(location.coordinate.latitude), \(location.coordinate.longitude)")
+            print("Value: \(road.proximity(of: location.coordinate)) < \(road.proximity(of: rawLocation.coordinate))")
+            locationUpdateExpectation.fulfill()
+        }
+        
+        func passiveLocationDataSource(_ dataSource: PassiveLocationDataSource, didUpdateHeading newHeading: CLHeading) {
+        }
+        
+        func passiveLocationDataSource(_ dataSource: PassiveLocationDataSource, didFailWithError error: Error) {
+        }
+    }
+    
+    func testManualLocations() {
+//        let directions = DirectionsSpy()
+        let locationManager = PassiveLocationDataSource()
+        
+        let startingExpectation = expectation(description: "Location manager should start without an error")
+        locationManager.startUpdatingLocation { (error) in
+            XCTAssertNil(error)
+            startingExpectation.fulfill()
+        }
+        wait(for: [startingExpectation], timeout: 2)
+        
+        // FIXME: Starting the location manager should automatically configure the navigator.
+        XCTAssertNoThrow(try locationManager.configureNavigator(withTilesVersion: "1234"))
+        
+        let locationUpdateExpectation = expectation(description: "Location manager should respond to every manual location update")
+        locationUpdateExpectation.expectedFulfillmentCount = 5
+        
+        let road = Road(from: CLLocationCoordinate2D(latitude: 47.207966, longitude: 9.527012), to: CLLocationCoordinate2D(latitude: 47.209518, longitude: 9.522167))
+        let delegate = Delegate(road: road, locationUpdateExpectation: locationUpdateExpectation)
+        locationManager.delegate = delegate
+        
+        DispatchQueue.main.asyncAfter(deadline: .now() + .seconds(10)) {
+            locationManager.updateLocation(CLLocation(latitude: 47.208674, longitude: 9.524650))
+        }
+        
+        DispatchQueue.main.asyncAfter(deadline: .now() + .seconds(25)) {
```

Comment:
Probably makes sense to review this test in future since it's time consuming.

Reply:
Yep, in fact, I disabled this test because it requires network access and a valid Mapbox access token to function at all. We’ll figure out a better integration test as tail work. I just rewrote history in this PR to undo the change that deleted the offline routing tests; I think we could possibly leverage that test’s test fixture to avoid reinventing the wheel.

Reply:
#2546